components/FormInput/SharedFormInput.js

```diff
@@ -60,12 +60,13 @@ export default class SharedFormInput extends Component {
       error,
       value,
       label,
+      labelOutside,
       ...remainingProps,
     } = this.props;
     const { focus, id } = this.state;
 
-    const active = value && !!value.length;
-    const isInsideVariant = !!placeholder;
+    const active = labelOutside ? !!placeholder : (value && !!value.length);
+    const isInsideVariant = labelOutside ? false : !!placeholder;
```

Comment:
Probably best to rename `active`. This logic isn't easy to follow now we have the extra complication of the `labelOutside` prop.